File: numeric/euclidgcd/euclidgcd.jl

function euclidgcd(m, n)
    r = m % n
    while r != 0
        m = n
        n = r
        r = m % n
    end
    return n
end

v_1 = 15
v_2 = 10

if length(ARGS) >= 2
    v_1 = parse(Int, ARGS[1])
    v_2 = parse(Int, ARGS[2])
end

println(v_1, " ", v_2)
println("gcd: ", euclidgcd(v_1, v_2))

File: numeric/euclidgcd/euclidgcd.dart

int euclidgcd(int m, int n) {
    int r = 0;
    while (n != 0) {
        r = m % n;
        m = n;
        n = r;
    }
    return m;
}

void main(List<String> arguments) {
  int m = 15;
  int n = 10;

  if (arguments.length >= 2) {
    m = int.parse(arguments[0]);
    n = int.parse(arguments[1]);
  }

  int gcd = euclidgcd(m, n);

  print("${m} ${n}");
  print("gcd: ${gcd}");
}
